File: CALICO/Spring24/contest/lavapit/templates/lavapit_template.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

/**
 * Return the maximum number of diamonds that Steve can mine before exiting the lava pit.
 * 
 * N: number of rows in the lava pit
 * M: number of columns in the lava pit
 * G: description of the lava pit
 */
int solve(int N, int M, vector<string> G) {
    // YOUR CODE HERE
    return -1;
}

int main() {
    int T;
    cin >> T;
    while (T--) {
        int N, M;
        cin >> N >> M;
        vector<string> G(N);
        for (int i = 0; i < N; ++i) {
            cin >> G[i];
        }
        cout << solve(N, M, G) << '\n';
    }
}

File: CALICO/Spring24/contest/lavapit/templates/lavapit_template.py
def solve(N: int, M: int, G: list[str]) -> int:
    """
    Return the maximum number of diamonds that Steve can mine before exiting the lava pit.

    N: number of rows in the lava pit
    M: number of columns in the lava pit
    G: description of the lava pit
    """
    # YOUR CODE HERE
    return 0


def main():
    T = int(input())
    for _ in range(T):
        N, M = map(int, input().split())
        G = [input() for i in range(N)]
        print(solve(N, M, G))


if __name__ == '__main__':
    main()
